src/otx/core/config/callbacks.py

```diff
@@ -0,0 +1,4 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Config data type objects for callbacks."""
```

Comment:
> Excuse me, but I can't understand, why you created these empty files. If we want to use https://hydra.cc/docs/tutorials/structured_config/schema/ then I assume we should create base_config for each group of yaml based configs. Shouldn't we?

First of all, you don't need to register each group's config DTO to `ConfigStore`. You can see here that `TrainConfig` is only one currently registered to validate the configuration schema.
https://github.com/openvinotoolkit/training_extensions/blob/61c09ab3c6022bd5ab2f697c71eb39927ab03f00/src/otx/core/config/__init__.py#L29-L34
This is possible since `TrainConfig` has hierarchy of children config DTOs, so that this makes Hydra to validate their schemas as well.

Secondly, the reason why the empty file exist to guide your implementation and let you know about the future direction (not only for this file, but every other empty file). For example, this PR currently doesn't have the training recipe which has callback configurations. However, while doing your P1 target, you might need to add callbacks. In that case, you can do something to this empty file.